rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -252,63 +297,98 @@ rclcpp::remove_ros_arguments(int argc, char const * const argv[])
 
   ret = rcl_arguments_fini(&parsed_args);
   if (RCL_RET_OK != ret) {
-    exceptions::throw_from_rcl_error(ret, "failed to cleanup parsed arguments, leaking memory");
+    rclcpp::exceptions::throw_from_rcl_error(
+      ret, "failed to cleanup parsed arguments, leaking memory");
   }
 
   return return_arguments;
 }
 
 bool
-rclcpp::ok()
+rclcpp::ok(rclcpp::Context::SharedPtr context)
 {
-  return ::g_signal_status == 0;
+  using rclcpp::contexts::default_context::get_global_default_context;
+  if (nullptr != context) {
+    context = get_global_default_context();
+  }
+  auto rcl_context_wrapper = context->get_sub_context<RclContextWrapper>();
+  return rcl_context_is_valid(rcl_context_wrapper->get_context().get());
 }
 
 bool
-rclcpp::is_initialized()
+rclcpp::is_initialized(rclcpp::Context::SharedPtr context)
 {
-  return rcl_ok();
+  using rclcpp::contexts::default_context::get_global_default_context;
+  if (nullptr != context) {
```

Comment:
ditto
```suggestion
  if (nullptr == context) {
```

Reply:
Maybe just replace this implementation with `return ok(context);` ?

Out of curiosity, when do you think is an appropriate time to remove deprecated functions, such as this one?

Reply:
So, there's been a long debate about `rclcpp::ok()`'s name, and I didn't really want to revisit it. `is_initialized()` was added to differentiate between ok (whether sigint was called or not) and whether or not init had been called. Now those are essentially the same thing, so there isn't a good reason for one or the other. Personally I'd drop `is_initialized()` since no one is probably using it very much yet, as opposed to `ok()` which is used in several examples and in rclcpp itself. This despite the fact that rcl calls it "is_initialized" and that we've discussed changing the `ok()` name before (though I'm fine with it).